server/environment.go

```diff
@@ -16,48 +16,52 @@ import (
 func (s *Server) registerEnvironmentRoutes(g *echo.Group) {
 	g.POST("/environment", func(c echo.Context) error {
 		ctx := context.Background()
-		environmentCreate := &api.EnvironmentCreate{}
-		if err := jsonapi.UnmarshalPayload(c.Request().Body, environmentCreate); err != nil {
+		envCreate := &api.EnvironmentCreate{}
+		if err := jsonapi.UnmarshalPayload(c.Request().Body, envCreate); err != nil {
 			return echo.NewHTTPError(http.StatusBadRequest, "Malformatted create environment request").SetInternal(err)
 		}
 
-		environmentCreate.CreatorID = c.Get(getPrincipalIDContextKey()).(int)
+		envCreate.CreatorID = c.Get(getPrincipalIDContextKey()).(int)
 
-		environment, err := s.EnvironmentService.CreateEnvironment(ctx, environmentCreate)
+		envRaw, err := s.EnvironmentService.CreateEnvironment(ctx, envCreate)
 		if err != nil {
 			if common.ErrorCode(err) == common.Conflict {
-				return echo.NewHTTPError(http.StatusConflict, fmt.Sprintf("Environment name already exists: %s", environmentCreate.Name))
+				return echo.NewHTTPError(http.StatusConflict, fmt.Sprintf("Environment name already exists: %s", envCreate.Name))
 			}
 			return echo.NewHTTPError(http.StatusInternalServerError, "Failed to create environment").SetInternal(err)
 		}
 
-		if err := s.composeEnvironmentRelationship(ctx, environment); err != nil {
+		env, err := s.composeEnvironmentRelationship(ctx, envRaw)
+		if err != nil {
 			return echo.NewHTTPError(http.StatusInternalServerError, "Failed to fetch created environment relationship").SetInternal(err)
 		}
 
 		c.Response().Header().Set(echo.HeaderContentType, echo.MIMEApplicationJSONCharsetUTF8)
-		if err := jsonapi.MarshalPayload(c.Response().Writer, environment); err != nil {
+		if err := jsonapi.MarshalPayload(c.Response().Writer, env); err != nil {
 			return echo.NewHTTPError(http.StatusInternalServerError, "Failed to marshal create environment response").SetInternal(err)
 		}
 		return nil
 	})
 
 	g.GET("/environment", func(c echo.Context) error {
 		ctx := context.Background()
-		environmentFind := &api.EnvironmentFind{}
+		envFind := &api.EnvironmentFind{}
 		if rowStatusStr := c.QueryParam("rowstatus"); rowStatusStr != "" {
 			rowStatus := api.RowStatus(rowStatusStr)
-			environmentFind.RowStatus = &rowStatus
+			envFind.RowStatus = &rowStatus
 		}
-		envList, err := s.EnvironmentService.FindEnvironmentList(ctx, environmentFind)
+		envRawList, err := s.EnvironmentService.FindEnvironmentList(ctx, envFind)
 		if err != nil {
 			return echo.NewHTTPError(http.StatusInternalServerError, "Failed to fetch environment list").SetInternal(err)
 		}
 
-		for _, environment := range envList {
-			if err := s.composeEnvironmentRelationship(ctx, environment); err != nil {
-				return echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to fetch environment relationship: %v", environment.Name)).SetInternal(err)
+		envList := make([]*api.Environment, 0, len(envRawList))
```

Comment:
var envList []*apiEnvironment

Reply:
The current code has the advantage of accurate capacity, which will not expand the list. If we change it to `var envList []*apiEnvironment`, list expansion may happen.

Reply:
It is okay. Don't need to optimize performance as it doesn't match much.
https://go.dev/blog/slices

Reply:
Fixed.
Actually there are like 30 occurrence of the pattern `list := make([]*api.Xxx, 0)`. Should we change them to `var list []*api.Xxx`?

Reply:
Yeah, please. We can fix other usages later.

https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices

Reply:
try to fix in https://github.com/bytebase/bytebase/pull/827